### ISSUE TITLE
Anim State Graph exit Time is still confusing users 

### ISSUE THREAD

**Issue by yaustar** [playcanvas/editor#892]
Related thread: https://forum.playcanvas.com/t/click-button-to-control-animation-playback/27643

It seems like no matter what the default is, we always get support on either:

- Using conditions to change states don't work because the default of Exit Time is 1 (see https://github.com/playcanvas/editor/issues/804)
- Animations don't fully play when Exit time is 0

We need to rethink on how we can present this to the user so that they can see that they need to change it to adapt to their needs

<img width="501" alt="image" src="https://user-images.githubusercontent.com/16639049/192734707-e4c03425-06fd-409e-b6d7-61a2d3d6908c.png">

We've previously discussed about putting it under Conditions so it's clear it's a gating attribute for the transition

I was also thinking about changing it so that the value be a drop down that is:

- Any
- Custom - and this would show a slider that can be changed from 0-1

Having both of these would allow us to default back to 'Any'/0 and hopefully still be clear that it needs to changed if you want the full animation to play.

There is also the question if Exit Time is a clear enough name for users who aren't familiar with animation systems?

**Comment by Maksims** [comment]
An additional issue to this: Exit Time is in the range of 0..1, but it does not take into account animation speed, so if animation speed is 0.5, so Exit Time should be 2, to mark the end of it. And if speed is 2.0, then Exit Time should be 0.5 top.
This often leads to issues, especially when Exit Time is used, but the unaware dev changes the speed of animation, and then it misbehaves.

**Comment by jiajasper** [comment]
Please see first pass on design here: https://www.figma.com/file/USwW0a6rU12PJkcz4wfwzg/PlayCanvas-Editor?node-id=1023%3A34542

The key changes here are:

1. We are introducing a checkbox toggle to enable/disable "Interruptible Anytime". By default this toggle is checked meaning the animation exit time is set to 0 by default. 

2. We will show the exit time setting below the checkbox in the disabled state. If the checkbox is unchecked, we enable the exit time setting below and set the exit time to 0.5 (middle). If the user drag the slider back to 0, checkbox is automatically checked and the slider will be disabled again. This is designed to clearly indicate and educate the relationship.

3. When no conditions are set, we will introduce a note below that educate the user that the transition can be activated instantly or at exit time (depending on the exit time set). This is designed to help educate users about the relationship between exit time and conditions.


![Image](https://user-images.githubusercontent.com/30747200/200385315-f2c72403-f8c6-48c2-95e3-01d29a09bdcd.png)

![Image](https://user-images.githubusercontent.com/30747200/200391722-e668bef0-0c72-4672-a61f-2c299cbc6d9f.png)

**Comment by yaustar** [comment]
Thinking on this, although I used the term, Interruptible Anytime doesn't sound quite right as technically it's not being interrupted as the transition hasn't started.

Renaming it to be 'Always Active' (referring to the transition being always active) and having it enabled sounds better in my head. When unticked, it would show the slider which could be named 'Animation Exit Time' or 'State Exit Time' so it's clear that it's referring to the animation/state timeline.

**Comment by jiajasper** [comment]
After discussion with @ellthompson  we have made changes to the copy and decided to go with "Has Exit Time" for the checkbox and "Transition Exit Time" for the 0-1 parameter. This is very similar to Unity which would benefit those we came from Unity. We have also made changes to the note copy as shown below.

![image](https://user-images.githubusercontent.com/30747200/200632046-a512d1d2-8b42-43b3-92a3-ada75ed3f292.png)

![image](https://user-images.githubusercontent.com/30747200/200632095-2081ffeb-ea34-464b-86e5-71f409ee07ce.png)

Figma Link here:https://www.figma.com/file/USwW0a6rU12PJkcz4wfwzg/PlayCanvas-Editor?node-id=1040%3A39460
